editor/src/clj/editor/material.clj

```diff
@@ -123,6 +124,25 @@
                          (format "'%s' attribute uses normalize with float data type"
                                  name)))]))
 
+(defn pb-parameter-type->pbr-parameter-type-pb-keyword [pbr-parameter-type]
+  (condp = pbr-parameter-type
+    MaterialBuilder$PbrParameterType/PBR_PARAMETER_TYPE_METALLIC_ROUGHNESS :has-metallic-roughness
+    MaterialBuilder$PbrParameterType/PBR_PARAMETER_TYPE_SPECULAR_GLOSSINESS :has-specular-glossiness
+    MaterialBuilder$PbrParameterType/PBR_PARAMETER_TYPE_CLEARCOAT :has-clearcoat
+    MaterialBuilder$PbrParameterType/PBR_PARAMETER_TYPE_TRANSMISSION :has-transmission
+    MaterialBuilder$PbrParameterType/PBR_PARAMETER_TYPE_IOR :has-ior
+    MaterialBuilder$PbrParameterType/PBR_PARAMETER_TYPE_SPECULAR :has-specular
+    MaterialBuilder$PbrParameterType/PBR_PARAMETER_TYPE_VOLUME :has-volume
+    MaterialBuilder$PbrParameterType/PBR_PARAMETER_TYPE_SHEEN :has-sheen
+    MaterialBuilder$PbrParameterType/PBR_PARAMETER_TYPE_EMISSIVE_STRENGTH :has-emissive-strength
+    MaterialBuilder$PbrParameterType/PBR_PARAMETER_TYPE_IRIDESCENCE :has-iridescene))
+
+(defn- build-target-pbr-params [shader-reflection]
+  (when-some [pbr-parameters (MaterialBuilder/getPbrParameters shader-reflection)]
+    (into {:has-parameters true}
+          (map #(vector (pb-parameter-type->pbr-parameter-type-pb-keyword %) true))
+          pbr-parameters)))
+
```

Comment:
Suggestion: Why not have a static function in Bob that takes a `Graphics.ShaderDesc.ShaderReflection` and returns a populated `Material.MaterialDesc.PbrParameters` protobuf message? Then both Bob and the editor could use that.

For the editor, you can convert a protobuf message to a Clojure map using `protobuf/pb->map-with-defaults` or `pb->map-without-defaults`, depending on if you want the default values for the fields written to the binaries or not. We typically don't write default field values, but I saw the test you added expects them for some reason.

Reply:
Not sure I understand, the test I wrote checks that all values have been set. I don't care about the default values, as they will be false.

Reply:
Hmm, I might have misread the test. 😅

Anyway, I think we're doing the correct thing here by not including default pb-field values in the built binaries, so everything is fine here.